### DISCUSSION TITLE
How to count the number of certain key value pair?

### DISCUSSION THREAD

**Discussion by sirlis** [nlohmann/json#3753] [Q&A]
I have a json file similar to:
```json
{
  "objects": [
      {
          "CategoryID": 0,
          "TypeID": 0,
          "parameter1": 5,
          "parameter2": 1,
      },
      {
          "CategoryID": 0,
          "TypeID": 1,
          "parameter1": 5,
          "parameter2": 1,
      },
      {
          "CategoryID": 0,
          "TypeID": 1,
          "parameter1": 5,
          "parameter2": 1,
      },
      {
          "CategoryID": 1,
          "TypeID": 0,
          "parameter1": 5,
          "parameter2": 1,
      }
    ]
}
```

Is it allowed to have duplicated elements (e.g., `CategoryID = 0 and TypeID = 1` in the above json file)?

If it is allowed, how to get the number of objects whose `CategoryID = 0 and TypeID = 1` (the number should be 2 according to the above json file). 

I searched for the function `json::count` but it only counts for `key`, not the key value pair.

**Comment by falbrechtskirchinger** [comment]
> Is it allowed to have duplicated elements (e.g., `CategoryID = 0 and TypeID = 1` in the above json file)?

Yes.

> If it is allowed, how to get the number of objects whose `CategoryID = 0 and TypeID = 1` (the number should be 2 according to the above json file).

There's no library function for what you want. You can either iterate through the array and count manually or write a predicate and use [`std::count_if()`](https://en.cppreference.com/w/cpp/algorithm/count).

**Reply by sirlis** [reply]
Thank you. I will try to write a predicate and use `std::count_if()`. I am new to c++ and I doubt if I can work it out though.  T.T
